python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -182,6 +255,124 @@ def __init__(self, *args, **kwargs):
+        output = Struct(
+            outputExposure=outputExposure,
+            outputBackgroundExposure=outputBackgroundExposure,
+            outputBadStamps=badStamps,
+        )
+        butlerQC.put(output, outputRefs)
+
+    def run(
+        self, inputExposure, inputBrightStarStamps, inputExtendedPsf, dataId, skyCorr=None, refObjLoader=None
+    ):
+        """Iterate over all bright stars in an exposure to scale the extended
+        PSF model before subtracting bright stars.
+
+        Parameters
+        ----------
+        inputExposure : `~lsst.afw.image.ExposureF`
+            The image from which bright stars should be subtracted.
+        inputBrightStarStamps :
+                `~lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            Set of stamps centered on each bright star to be subtracted,
+            produced by running
+            `~lsst.pipe.tasks.processBrightStars.ProcessBrightStarsTask`.
+        inputExtendedPsf : `~lsst.pipe.tasks.extended_psf.ExtendedPsf`
+            Extended PSF model, produced by
+            `~lsst.pipe.tasks.extended_psf.MeasureExtendedPsfTask`.
+        dataId : `dict` or `~lsst.daf.butler.DataCoordinate`
+            The dataId of the exposure (and detector) bright stars should be
+            subtracted from.
+        skyCorr : `~lsst.afw.math.backgroundList.BackgroundList`, optional
+            Full focal plane sky correction, obtained by running
+            `~lsst.pipe.tasks.skyCorrection.SkyCorrectionTask`. If
+            `doApplySkyCorr` is set to `True`, `skyCorr` cannot be `None`.
+        refObjLoader : `~lsst.meas.algorithms.ReferenceObjectLoader`, optional
+            Loader to find objects within a reference catalog.
+
+        Returns
+        -------
+        subtractorExp : `~lsst.afw.image.ExposureF`
+            An Exposure containing a scaled bright star model fit to every
+            bright star profile; its image can then be subtracted from the
+            input exposure.
+        invImages : `list` [`~lsst.afw.image.MaskedImageF`]
+            A list of small images ("stamps") containing the model, each scaled
+            to its corresponding input bright star.
+        """
+        self.inputExpBBox = inputExposure.getBBox()
+        if self.config.doApplySkyCorr and (skyCorr is not None):
+            self.log.info(
+                "Applying sky correction to exposure %s (exposure will be modified in-place).", dataId
+            )
+            self.applySkyCorr(inputExposure, skyCorr)
+
+        # Create an empty image the size of the exposure.
+        # TODO: DM-31085 (set mask planes).
+        subtractorExp = ExposureF(bbox=inputExposure.getBBox())
+        subtractor = subtractorExp.maskedImage
+
+        # Make a copy of the input model.
+        self.model = inputExtendedPsf(dataId["detector"]).clone()
+        self.modelStampSize = self.model.getDimensions()
+        self.inv90Rots = 4 - inputBrightStarStamps.nb90Rots % 4
```

Comment:
This line could use a comment. I know what the 90rots stuff is now but not what this does.